packages/protocol/test/governance/network/governance.ts

```diff
@@ -1608,6 +1608,47 @@ contract('Governance', (accounts: string[]) => {
     })
   })
 
+  describe.only('#unapprove()', () => {
+    const proposalId = 1
+    const index = 0
+    beforeEach(async () => {
+      await governance.propose(
+        [transactionSuccess1.value],
+        [transactionSuccess1.destination],
+        // @ts-ignore bytes type
+        transactionSuccess1.data,
+        [transactionSuccess1.data.length],
+        descriptionUrl,
+        // @ts-ignore: TODO(mcortesi) fix typings for TransactionDetails
+        { value: minDeposit }
+      )
+      await timeTravel(dequeueFrequency, web3)
+      await governance.approve(proposalId, index)
+    })
+
+    it('should return true', async () => {
+      const success = await governance.unapprove.call(proposalId, index)
+      assert.isTrue(success)
+    })
+
+    it('should set the proposal to unapproved', async () => {
+      await governance.unapprove(proposalId, index)
+      assert.isFalse(await governance.isApproved(proposalId))
+    })
```

Comment:
can we add a test for reverting `execute()` when in `Execution` phase and passing but unapproved

Reply:
Added tests to cover execute() after proposal is unapproved during execution stage.